nuclio/auth.py

```diff
@@ -0,0 +1,36 @@
+import os
+
+import requests.auth
+
+
+class AuthInfo(requests.auth.HTTPBasicAuth):
+    """
+    Accommodate Iguazio-based authentication mechanism
+    """
+
+    @classmethod
+    def from_session_key(cls, session_key):
+        return cls("", session_key)
+
+    @classmethod
+    def from_envvar(cls):
+        access_key = os.getenv("V3IO_ACCESS_KEY")
+        if access_key:
+            return cls.from_session_key(access_key)
+
+        return None
```

Comment:
After reading the rest of the code what I would do is:
* create a regular class named `IguazioAuth` or `IguazioAuthInfo`
* Instead of inheritance simply add a method `to_requests_auth` which then may return None (still a bit weird but less), BTW what will happen if I'm initializing a `HTTPBasicAuth` class instance with 2 empty values ? will it fail the request ? 

The value is:
* We don't tie ourselves to requests Auth classes structure (maybe we will use other package at some point, you don't want that to cause a change in your interfaces)
* We may want to use this `AuthInfo` class for passing around more variables and then the 2 variables  `HTTPBasicAuth` has won't be enough

Reply:
adapted some of the suggestions above, `AuthInfo` would behave like a factory now, provided by "mode" which would be auto populated when using `from_envvar`